src/OrchardCore.Modules/OrchardCore.Media/Deployment/MediaDeploymentSource.cs

```diff
@@ -37,36 +29,35 @@ public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlan
                    select fileStoreEntry.Path).ToArray()
                 : mediaStep.Paths;
 
-            var files = new List<JObject>(paths.Length);
-
             foreach (var path in paths)
             {
-                var base64 = await ReadFileAsBase64Async(path);
-
-                var file = new JObject(
-                    new JProperty("Path", path),
-                    new JProperty("Base64", base64));
+                var content = await ReadFileAsync(path);
 
-                files.Add(file);
+                await result.FileBuilder.SetFileAsync(path, content);
             }
 
             // Adding media files
             result.Steps.Add(new JObject(
                 new JProperty("name", "media"),
-                new JProperty("Files", JArray.FromObject(files.ToArray()))
-            ));
+                new JProperty("Files", JArray.FromObject((from path in paths
+                                                          select new JObject(
+                                                              new JProperty("Path", path),
+                                                              new JProperty("Base64", $"[file:base64('{path}')]")
```

Comment:
This is bad also. I know I implemented it, but I was young. We need another property on the "media" step that represents a __source__ path. If set we just need to get the stream. The `RecipeExecutionContext` has a `FileProvider` to read the file. This way we can just stream from the source to the destination without loading the file in memory (and converting to base64 either). That will change the steps, and we can also update the recipes to use it, setup will be faster. But we can keep Base64 in case we really want to embed a file content and not point to the actual file.

Reply:
Arrgghh ... I keep forgetting your comment that "we are gods of the code, not the other way around" ... and I actually had that impl. and then backed it out ... so silly of me. I'll put it back in.

Reply:
Done.